host/xtest/regression_1000.c

```diff
@@ -2349,3 +2349,214 @@ static void xtest_tee_test_1032(ADBG_Case_t *c)
 }
 ADBG_CASE_DEFINE(regression, 1032, xtest_tee_test_1032,
 		"Register read-only shared memory");
+
+/*
+ * regression_1033
+ * Depends on kernel commit "tee: add support for application-based session login methods"
+ * Mailing list archives: https://lists.trustedfirmware.org/pipermail/op-tee/2020-October/000211.html
+ * Upstream: <put sha-1 here when known>
+ *
+ * xtest skips the test when not built with OpenSSL.
+ */
+static void xtest_tee_test_1033(ADBG_Case_t *c)
+{
+#ifdef OPENSSL_FOUND
+	TEEC_Result result = TEEC_ERROR_GENERIC;
+	uint32_t ret_orig = 0;
+	TEEC_Session session = { };
+	uint32_t login = UINT32_MAX;
+	TEEC_UUID client_uuid = { };
+	TEEC_UUID expected_client_uuid = { };
+	TEEC_UUID uuid_ns = { };
+	char uuid_name[TEE_UUID_NS_NAME_SIZE + 4] = { };
+	char app[PATH_MAX] = { };
+
+	result = xtest_uuid_from_str(&uuid_ns, client_uuid_linux_ns);
+
+	if (!ADBG_EXPECT_TEEC_SUCCESS(c, result))
+		return;
+
+	if (readlink("/proc/self/exe", app, PATH_MAX) < 0) {
+		Do_ADBG_Log(" - 1033 -   skip test, "
+			    "/proc/self/exe not readable");
+		return;
+	}
+
+	snprintf(uuid_name, TEE_UUID_NS_NAME_SIZE + 4, "app=%s", app);
```

Comment:
s/TEE_UUID_NS_NAME_SIZE + 16/sizeof(uuid_name)/

Also check truncation, I.e.:
```c
	len = snprintf(uuid_name, sizeof(uuid_name),  "app=%s", app);
	if (ADBG_EXPECT_TRUE(c, len > 0 && len < sizeof(uuid_name)))
		return;
```

Last, can remove line below (`uuid_name[TEE_UUID_NS_NAME_SIZE] = 0;`) as stated by @jenswi-linaro .

Reply:
[A previous comment](https://github.com/OP-TEE/optee_test/pull/468#discussion_r533156066) explains why we need a manual truncation at `TEE_UUID_NS_NAME_SIZE`.

The return value of snprintf cannot be equal to or more than `sizeof(uuid_name)` in this case since `sizeof(uuid_name)` is defined as length of `app` plus four. I would not trust my own judgement on this, but the compiler agrees, since if the truncation *was* possible, the compiler would refuse to work with the `‘snprintf’ output between 5 and 4100 bytes into a destination of size 4096`-type errors as shown [above](https://github.com/OP-TEE/optee_test/pull/468#discussion_r533156066).

I will replace hard-coded lengths with `sizeof(uuid_name)`. I don't know cases in which `snprintf` can return a negative number that don't include your system catching on fire but I will add a check for this.